.github/workflows/llvm-ci.yml

```diff
@@ -0,0 +1,246 @@
+name: Build LLVM (matrix)
+
+on:
+  workflow_dispatch:
+    inputs:
+      llvm_pin:
+        description: "Commit SHA, tag (e.g. llvmorg-21.1.1), or ref for llvm/llvm-project"
+        required: false
+        default: llvmorg-21.1.1
+  push:
+    branches: [ main ]
+  pull_request:
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 180
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-22.04, ubuntu-22.04-arm, macos-15, windows-2025]
+
+    env:
+      # manual input → repo/org var → default
+      LLVM_PIN: ${{ github.event.inputs.llvm_pin || vars.LLVM_PIN || 'llvmorg-21.1.1' }}
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
```

Comment:
Let's use python 3.10 please. It's the minimum version we are supporting in gstaichi currently. prefer to stadnardize across CI